### ISSUE TITLE
Column chooser accessibility issue with tab index

### ISSUE THREAD

**Issue by svetq** [telerik/kendo-angular#4831]
### Description

In the column chooser, all hidden elements have a tab index, causing accessibility issues.

### Steps To Reproduce

Open the following [demo](https://www.telerik.com/kendo-angular-ui/components/grid/accessories/column-menu#using-standalone-column-chooser) in a new window. Open the column-chooser, then run Axe accessibility tool on the page to see the errors.

### Actual Behavior

There is an "Interactive controls must not be nested" error

### Expected Behavior

There should be no "Interactive controls must not be nested" error

### Browser

Chrome

### Browser version

latest

### OS type

MacOS

**Comment by glnlcu** [comment]
Also on [this page](https://www.telerik.com/kendo-angular-ui/components/grid/accessories/column-menu#using-standalone-column-chooser), I noticed that in the columnChooser, if a column is hidden because [includeInChooser]="false" is set, it still becomes visible in the list when trying to filter. Specifically, if I type 'c', the 'Carrier' column appears. Does this issue also fix this bug?"

**Comment by FloppyNotFound** [comment]
I'm experiencing the same issue with hidden columns appearing in the search results.
Here's an independent StackBlitz, try searching for "Category":

https://stackblitz.com/edit/angular-ul6tmedh?file=src%2Fapp%2Fapp.component.ts
